src/test/java/com/casper/sdk/service/CasperServiceTests.java

```diff
@@ -338,6 +345,23 @@ void queryGlobalState() {
         assertNotNull(globalState);
     }
 
+    @Test
+    void queryBalance() throws IOException {
+        final URL privateKeyAccount1 = getClass().getResource("/deploy-accounts/nctl/users/user-1/secret_key.pem");
+        Assertions.assertNotNull(privateKeyAccount1);
+        final AbstractPrivateKey privateKey = new Ed25519PrivateKey();
+        privateKey.readPrivateKey(privateKeyAccount1.getFile());
+        PublicKey publicKeyAccount1 = PublicKey.fromAbstractPublicKey(privateKey.derivePublicKey());
+        GlobalStateIdentifier stateRootHashIdentifier = StateRootHashIdentifier.builder()
+                .hash(casperServiceNctl.getStateRootHash().getStateRootHash())
+                .build();
+        PurseIdentifier mainPurseUnderPublickey = MainPurseUnderPublickey.builder()
+                .publicKey(publicKeyAccount1)
+                .build();
+        QueryBalanceData balanceData = casperServiceNctl.queryBalance(stateRootHashIdentifier, mainPurseUnderPublickey);
+        assertNotNull(balanceData);
```

Comment:
Perhaps we should also validate the contents of QueryBalanceData.

Reply:
Good idea. I'll check if it is non zero, since other transfers can take place before my test, so I can't be sure of the exact balance.